### PR TITLE
Disregard dtx when pruning IC history

### PR DESCRIPTION
**Pull request by xuebinsu**
This patch aims to fix the memory leak issue #10314 .

Previously, the IC history table got pruned only after begining a
new distributed transaction (dtx). As a result, 

- when a transaction contains many commands, or 
- when a session never begins any distributed transaction, 

the history will never get pruned. This can cause memory leak.

This patch fixes this issue by reverting this part of code to the
last version before dependency on distributed transaction was
introduced. After that, we will only maintain a constant-length IC 
history. This ensures that

- The history table takes only constant amount of memory, and
- The history is long enough to handle mismatched packets.

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [x] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

**Comment by xuebinsu** [comment]
Status: Doing `make installcheck`

**Comment by xuebinsu** [comment]
Status: `make installcheck` passed.

**Review by ashwinstar — APPROVED** [review]
LGTM

File: src/backend/cdb/motion/ic_udpifc.c [hunk 1]
```diff
@@ -282,13 +282,6 @@ struct ReceiveControlInfo
 
 	/* Cursor history table. */
 	CursorICHistoryTable cursorHistoryTable;
-
-	/*
-	 * Last distributed transaction id when SetupUDPInterconnect is called.
-	 * Coupled with cursorHistoryTable, it is used to handle multiple
-	 * concurrent cursor cases.
-	 */
-	DistributedTransactionId lastDXatId;
```
Comment (Aegeaner) [inline src/backend/cdb/motion/ic_udpifc.c:291]: After removing this, may need to hold a lock on `rx_control_info.cursorHistoryTable` to handle multiple concurrent cursor cases ?

Comment (Aegeaner) [reply src/backend/cdb/motion/ic_udpifc.c:291]: There is already a mutex hold in https://github.com/greenplum-db/gpdb/blob/ffb2a72850935b160a1f4058e5eb71892bf3751b/src/backend/cdb/motion/ic_udpifc.c#L3014 , ignore above.

**Review by Aegeaner — APPROVED** [review]

**Comment by xuebinsu** [comment]
Thanks for reviewing. Merging.

**Pull request merged**